enterprise/internal/batches/sources/bitbucketcloud_test.go

```diff
@@ -0,0 +1,889 @@
+package sources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	bbcs "github.com/sourcegraph/sourcegraph/enterprise/internal/batches/sources/bitbucketcloud"
+	btypes "github.com/sourcegraph/sourcegraph/enterprise/internal/batches/types"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/auth"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/bitbucketcloud"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+func TestNewBitbucketCloudSource(t *testing.T) {
+	t.Run("invalid", func(t *testing.T) {
+		for name, input := range map[string]string{
+			"invalid JSON":    "invalid JSON",
+			"unparsable JSON": `{"appPassword": ["not a string"]}`,
```

Comment:
```suggestion
			"invalid schema": `{"appPassword": ["not a string"]}`,
```

I'm sorry, super nitpicky. This is technically parseable JSON.

Reply:
![48c](https://user-images.githubusercontent.com/229984/164760696-6b1ad5c8-deb5-4d71-8a32-7900b282cecb.jpg)